packages/DApp/src/hooks/useUnverifiedVotes.ts

```diff
@@ -38,33 +39,49 @@ export function useUnverifiedVotes(room: number | undefined, verificationStartAt
     const accumulateVotes = async () => {
       if (waku && room) {
         const messages = await wakuMessage.receive(waku, config.wakuConfig.wakuTopic, room)
-        const validMessages = messages?.filter((message) => validateVote(message, verificationStartAt, startAt))
 
-        const votes: InitialVotes =
-          validMessages?.reduce((acc, message) => {
-            if (acc.voted.includes(message.address)) {
-              return { for: acc.for, against: acc.against, voted: acc.voted }
-            }
+        if (!messages?.length) {
+          return
+        }
+
+        const validMessages = messages.filter((message) => validateVote(message, verificationStartAt, startAt))
+
+        if (!validMessages.length) {
+          return
+        }
 
-            if (message.vote === 'no') {
-              return {
-                for: acc.for,
-                against: acc.against + parseInt(message.sntAmount._hex, 16),
-                voted: [...acc.voted, message.address],
-              }
-            } else if (message.vote === 'yes') {
-              return {
-                for: acc.for + parseInt(message.sntAmount._hex, 16),
-                against: acc.against,
-                voted: [...acc.voted, message.address],
-              }
+        const votes: InitialVotes = validMessages.reduce((acc, message) => {
+          // todo?: if non-proposing accounts can vote multiple times
+          // account already voted
+          // if (acc.voted.includes(message.address)) {
+          //   // fixme?: do not include votes of the proposing account
+          //   console.log('ALREADY VOTED')
+          //   return { for: acc.for, against: acc.against, voted: acc.voted }
+          // }
```

Comment:
⚠️

Reply:
You are correct. Non-proposing accounts can vote multiple times, but it should count only last vote.

Reply:
So are you okay with leaving this condition commented out? Since we're not disallowing to vote multiple times? And would only the last vote count for the proposing account as well then? Essentially allowing editing on the proponent's end. I haven't checked the contract yet.

Reply:
Based on https://github.com/status-im/community-dapp/blob/e6e28dbf6da1d69908c0f065ad2d7991551a7d60/packages/contracts/contracts/VotingContract.sol#L285-L288 I'd say only anyone's first vote counts.

Reply:
Everyone can vote just once, the proposing account included. Currently, for algorithm simplification (to save gas cost) only first vote counts.

Reply:
> Everyone can vote just once, the proposing account included.

In UI, addressed at https://github.com/status-im/community-dapp/pull/31/commits/528f8b60cf968f57f54312304e33cc272785ed17

> only first vote counts.

In UI, follow-up at https://github.com/status-im/community-dapp/issues/35.